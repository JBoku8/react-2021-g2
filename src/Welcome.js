const Welcome = (props) => {
  const styles = {
    color: "seagreen",
    textAlign: "center",
  };
  return (
    <div>
      <h1 title="Title example" style={styles}>
        {props.message}
      </h1>
    </div>
  );
};

export default Welcome;
