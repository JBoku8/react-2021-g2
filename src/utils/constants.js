export const firstName = 'John'; // named export
export const lastName = 'Doe'; // named export

const fullName = `${firstName} ${lastName}`;
export default fullName; // default export
