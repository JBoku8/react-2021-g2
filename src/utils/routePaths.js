export const TODO_PATH = '/todo-list';
export const COUNTER_PATH = '/counter';
export const PROFILE_PATH = '/profile';
export const FAKER_BOOKS_PATH = '/faker-books';
export const REDUX_COUNTER_PATH = '/redux-counter';
export const HOME_PATH = '/';

export const AUTH_PATH = '/auth';
export const SIGN_IN_PATH = '/auth/sign-in';
export const SIGN_UP_PATH = '/auth/sign-up';
