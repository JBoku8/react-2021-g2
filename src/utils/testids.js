export const TEST_IDS = {
  welcome: {
    title: 'welcome.title',
    addButton: 'welcome.addButton',
    resetButton: 'welcome.resetButton',
    subtractButton: 'welcome.subtractButton',
  },
  greeting: 'greeting.text',
  parentComponent: {},
};
