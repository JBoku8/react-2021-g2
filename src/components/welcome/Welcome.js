import { useState } from 'react';
import { TEST_IDS } from '../../utils/testids';
import Greeting from '../greeting';
import { Button } from '../ui';
import Limitation from './Limitation';
// import { firstName as username } from "../../utils/constants";

const Welcome = ({ message, startValue = 0 }) => {
  const [value, setValue] = useState(message);
  const [counter, setCounter] = useState(startValue);

  const styles = {
    color: 'seagreen',
    textAlign: 'center',
  };
  // newValue !== oldValue
  // newObject !== oldObject
  const onClickHandler = () => {
    setValue('state updated');
  };

  const onAdd = () => {
    setCounter(counter + 1);
  };
  const onMinus = () => {
    setCounter(counter - 1);
  };
  const onReset = () => {
    setCounter(0);
  };

  // conditional rendering example
  if (counter === 10) {
    return <Limitation onClick={onReset} message="Limit has been reached" />;
  }

  return (
    <div className="">
      <h1
        title="Title example"
        style={styles}
        data-testid={TEST_IDS.welcome.title}>
        {value}
      </h1>
      <Greeting text={counter} title={`Current counter value is ${counter}`} />
      <Button
        onClick={onReset}
        text="Reset Counter"
        type="default"
        data-testid={TEST_IDS.welcome.resetButton}
      />
      <Button
        onClick={onAdd}
        text="Add value"
        type="danger"
        data-testid={TEST_IDS.welcome.addButton}
      />
      <Button
        onClick={onMinus}
        text="Subtract value"
        type="warning"
        data-testid={TEST_IDS.welcome.subtractButton}
      />
      <Button onClick={onClickHandler} type="danger" />
    </div>
  );
};

export default Welcome;
