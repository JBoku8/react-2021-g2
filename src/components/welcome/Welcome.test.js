import { render } from '@testing-library/react';
import { TEST_IDS } from '../../utils/testids';
import Welcome from './Welcome';

const mockProps = {
  message: 'mock-message',
  startValue: 0,
};
describe('Welcome component testing', () => {
  //   let component;

  //   beforeEach(() => {
  //     component = render(<Welcome message={mockMessage} />);
  //   });

  it('should render message', () => {
    const component = render(<Welcome {...mockProps} />);

    expect(component.getByTestId(TEST_IDS.welcome.title)).toBeInTheDocument();
    expect(component.getByTestId(TEST_IDS.welcome.title)).toHaveTextContent(
      mockProps.message,
    );
  });

  it('should increase number by one', () => {
    const component = render(<Welcome {...mockProps} />);
    const addButton = component.getByTestId(TEST_IDS.welcome.addButton);
    const greeting = component.getByTestId(TEST_IDS.greeting);
    addButton.click();

    expect(greeting).toHaveTextContent(mockProps.startValue + 1);
  });
});
