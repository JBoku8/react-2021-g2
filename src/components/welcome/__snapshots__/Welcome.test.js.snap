// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome snapshots renders successfully 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
      >
        <h1
          data-testid="welcome.title"
          style="color: seagreen; text-align: center;"
          title="Title example"
        >
          mock-message
        </h1>
        <div>
          <h2
            class="text-muted fs-1"
            data-testid="greeting.text"
            title="Current counter value is 0"
          >
            0
          </h2>
        </div>
        <button
          class="btn mx-1 default "
          data-testid="welcome.resetButton"
          type="button"
        >
          Reset Counter
        </button>
        <button
          class="btn mx-1 danger "
          data-testid="welcome.addButton"
          type="button"
        >
          Add value
        </button>
        <button
          class="btn mx-1 warning "
          data-testid="welcome.subtractButton"
          type="button"
        >
          Subtract value
        </button>
        <button
          class="btn mx-1 danger "
          type="button"
        >
          default text
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
    >
      <h1
        data-testid="welcome.title"
        style="color: seagreen; text-align: center;"
        title="Title example"
      >
        mock-message
      </h1>
      <div>
        <h2
          class="text-muted fs-1"
          data-testid="greeting.text"
          title="Current counter value is 0"
        >
          0
        </h2>
      </div>
      <button
        class="btn mx-1 default "
        data-testid="welcome.resetButton"
        type="button"
      >
        Reset Counter
      </button>
      <button
        class="btn mx-1 danger "
        data-testid="welcome.addButton"
        type="button"
      >
        Add value
      </button>
      <button
        class="btn mx-1 warning "
        data-testid="welcome.subtractButton"
        type="button"
      >
        Subtract value
      </button>
      <button
        class="btn mx-1 danger "
        type="button"
      >
        default text
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
